fenixedu-ist-giaf-invoices/src/main/java/pt/ist/fenixedu/giaf/invoices/EventProcessor.java

```diff
@@ -140,7 +140,9 @@ private static DebtExemption findDebtExemptionfor(final DebtInterestCalculator c
             if (accountingEntry instanceof DebtExemption && previousAccountingEntry == refund) {
                 return (DebtExemption) accountingEntry;
             }
-            previousAccountingEntry = accountingEntry;
+            if (accountingEntry instanceof Refund) {
```

Comment:
we only want to consider if it's of type Refund, because there can be different types of exemptions before the debtExemption and the really importante previous entry relevant for this method is if it is a Refund and if it is the same